lua/orgmode/agenda/init.lua

```diff
@@ -88,45 +89,50 @@ end
 
 function Agenda:prompt()
   self.filters:reset()
-  return utils.menu('Press key for an agenda command', {
-    {
-      label = 'Agenda for current week or day',
-      key = 'a',
-      action = function()
-        return self:agenda()
-      end,
-    },
-    {
-      label = 'List of all TODO entries',
-      key = 't',
-      action = function()
-        return self:todos()
-      end,
-    },
-    {
-      label = 'Match a TAGS/PROP/TODO query',
-      key = 'm',
-      action = function()
-        return self:tags()
-      end,
-    },
-    {
-      label = 'Like m, but only TODO entries',
-      key = 'M',
-      action = function()
-        return self:tags({ todo_only = true })
-      end,
-    },
-    {
-      label = 'Search for keywords',
-      key = 's',
-      action = function()
-        return self:search()
-      end,
-    },
-    { label = 'Quit', key = 'q' },
-    { label = '', separator = ' ', length = 1 },
-  }, 'Press key for an agenda command')
+  local menu = Menu:new({
+    title = 'Press key for an agenda command',
+    prompt = 'Press key for an agenda command',
+  })
+
+  menu:add_option({
```

Comment:
I think we can go with multiple arguments here instead of single table. I don't expect this to grow.
```lua
menu:add_option('Agenda for current week or day', 'a', function() 
  return self:agenda() 
end)
```

Same thing for separators.

Reply:
I think tables improve readability. If I read the code for the first time, I would not understand what the first and second lines are responsible for.

Reply:
I agree it does. Generally, I also prefer tables when the number of args goes over 3 or if more than 1 is optional. Here we must provide the first two, or even all of them.
Also, documenting them via annotations is very helpful for DX.

Since this is a matter of preference, I'll let @jgollenz give his vote on this and we'll go from there.

Reply:
Since `MenuOption` has proper annotations, I'm in favor of the tables. I agree, it does make things a bit clearer